content/class/Post.md

```diff
@@ -0,0 +1,316 @@
+
+```c#
+public Post.from_json (Json.Object obj)
+```
+
+Creates a new object out of a given Json object from other points.
+
+## Methods
+
+### `has_flag`
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public bool has_flag (PostFlag flag)
+```
+
+Returns if the Post has a specific flag set.
+
+### `set_flag`
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public void set_flag (PostFlag flag)
+```
+
+Sets a flag for this Post.
+
+### `unset_flag`
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public void unset_flag (PostFlag flag)
+```
+
+Removes a flag from this Post.
+
+## Fields
+
+### `flags` {#field_flags}
+
+```c#
+private Gee.HashSet<PostFlag> flags
+```
+
+Contains flags set for different states of an Post. See [`PostFlag`](../enum/PostFlag.md).
+
+### `raw_text` {#field_raw_text}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+private string raw_text
+```
+
+The [`text`](#property_text) as it is recieved from the API, to be later formatted.
+
+> This field might be only available in debug builds, as we should not need this field after formatting.
+
+| API   | Twitter 1    | Twitter 2    | Mastodon    |
+| ----- | ------------ | ------------ | ----------- |
+| Field | `text`       | `text`       | `content`   |
+| Type  | UTF-8 string | UTF-8 string | HTML string |
+
+### `text_entities` {#field_text_entities}
+
+![API: Twitter 1](https://img.shields.io/badge/API-Twitter%201-lightgrey?style=flat-square) ![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+private TextEntity[] text_entities
+```
+
+All entities found in the text, to provide additional informations for the text formatting. See [TextEntity](../structure/TextEntity.md).
+
+> This field might be only available in debug builds, as we should not need this field after formatting.
```

Comment:
We need text entities (and the raw text) when recreating the formatted text - e.g. when changing whether we hide trailing image URLs etc. I don't know how (or if) Mastadon could support that without a lot of very messy HTML manipulation.